bool.go

```diff
@@ -9,8 +9,9 @@ import (
 
 // Arrayb is an n-dimensional array of boolean values
 type Arrayb struct {
-	shape        []uint64
-	strides      []uint64
+	shape        []int
+	strides      []int
+	offset       []int
```

Comment:
As with the note in Array64, let's separate the addition of the `offset` field into its own branch/PR.